xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/internal/DefaultBeanDescriptor.java

```diff
@@ -286,6 +286,22 @@ private void setCommonProperties(DefaultPropertyDescriptor desc, Map<Class, Anno
             }
             group.setFeature(parameterFeature.value());
         }
+
+
+        PropertyDisplayType displayTypeAnnotation = (PropertyDisplayType) annotations.get(PropertyDisplayType.class);
+        Type displayType;
+        if (displayTypeAnnotation != null && displayTypeAnnotation.value().length > 0) {
+            Class[] types = displayTypeAnnotation.value().clone();
+            ArrayUtils.reverse(types);
```

Comment:
I don't understand what reverse is doing here. You should take the first element as raw type (List in your uni test) and then create a sub table with the rest for the parameters.

Reply:
The idea was as follow:
```
@PropertyDisplayType({Class1.class, Class2.class, Class3.class})
```
=> `Class1<Class2<Class3>>>`

Reply:
OK. For me `Class1<Class2<Class3>>>` is way more rare than `Class1<Class2, Class3>`.